samples/detect.js

```diff
@@ -23,24 +23,40 @@ const {Transform, pipeline} = require('stream');
 const pump = util.promisify(pipeline);
 
 function detectTextIntent(projectId, sessionId, queries, languageCode) {
+  const PROJECT_ID = projectId;
+  const SESSION_ID = sessionId;
+  const QUERIES = queries;
+  const LANGUAGE_CODE = languageCode;
   // [START dialogflow_detect_intent_text]
+
+  /**
+   * TODO(developer): UPDATE these variables before running the sample.
+   */
+  // const PROJECT_ID = 'PROJECT_ID';
+  // const SESSION_ID = String(Date.now()); // sessionId can be a random number or some type of user identifier (preferably hashed)
+  // const QUERIES = [
+  //   'Reserve a meeting room in Toronto office, there will be 5 of us',
+  //   'Next monday at 3pm for 1 hour, please', // Tell the bot when the meeting is taking place
+  //   'B'  // Rooms are defined on the Dialogflow agent, default options are A, B, or C
+  // ]
+  // const LANGUAGE_CODE = 'en';
+
   // Imports the Dialogflow library
   const dialogflow = require('dialogflow');
 
   // Instantiates a session client
   const sessionClient = new dialogflow.SessionsClient();
 
-  if (!queries || !queries.length) {
-    return;
-  }
+  async function detectIntent(
+    projectId,
```

Comment:
Most of these arguments are available from the outer wrapping function, correct?

Reply:
Most of the parameters, yes! But `query` param would be different for every request since the purpose of this function is to detect the intent of a given query.
We could simplify the function and have the function use a global variables (such as projectId, sessionId and languageCode) but I think this function signature gives better reusability and decoupling to the code.